File: csharp/isbn-verifier/IsbnVerifier.cs
public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        string cleanedIsbn = number.Replace("-", "").ToUpper();
        if (cleanedIsbn.Length != 10)
        {
            return false;
        }
        int sum = 0;
        for (int i = 0; i < 9; i++)
        {
            if (!char.IsDigit(cleanedIsbn[i]))
            {
                return false;
            }
            sum += (cleanedIsbn[i] - '0') * (10 - i);
        }
        char lastChar = cleanedIsbn[9];
        if (char.IsDigit(lastChar))
        {
            sum += lastChar - '0';
        }
        else if (lastChar == 'X')
        {
            sum += 10;
        }
        else
        {
            return false;
        }
        return sum % 11 == 0;
    }
}

File: typescript/isbn-verifier/isbn-verifier.ts
export function isValid(isbn: string): boolean {
  const cleanIsbn: string = isbn.replace(/-/g, '');
  if (!/^\d{9}[\dX]$/.test(cleanIsbn)) {
    return false;
  }
  const total: number = [...cleanIsbn].reduce((sum: number, char: string, index: number): number => {
    const value: number = char === 'X' ? 10 : parseInt(char, 10);
    return sum + value * (10 - index);
  }, 0);
  return total % 11 === 0;
}
